### PR TITLE
[release] Retry 0.2.129

### PR DESCRIPTION
**Pull request by ehigham**
Fix exporting of env vars in build.yaml for `release`. 
Make `release`:
- depend on tests
- use tested wheel (dont rebuild).

### CODE REVIEW

File: build.yaml [hunk 1]
```diff
@@ -3357,35 +3357,34 @@ steps:
...
+      GITHUB_OAUTH_HEADER_FILE=/io/github-oauth \
+      HAIL_GENETICS_HAIL_IMAGE=docker://{{ hailgenetics_hail_image.image }} \
+      HAIL_GENETICS_HAIL_IMAGE_PY_3_10=docker://{{ hailgenetics_hail_image_python_3_10.image }} \
+      HAIL_GENETICS_HAIL_IMAGE_PY_3_11=docker://{{ hailgenetics_hail_image_python_3_11.image }} \
+      HAIL_GENETICS_HAILTOP_IMAGE=docker://{{ hailgenetics_hailtop_image.image }} \
+      HAIL_GENETICS_VEP_GRCH37_85_IMAGE=docker://{{ hailgenetics_vep_grch37_85_image.image }} \
+      HAIL_GENETICS_VEP_GRCH38_95_IMAGE=docker://{{ hailgenetics_vep_grch38_95_image.image }} \
+      AZURE_WHEEL=/io/azure-wheel/hail-*-py3-none-any.whl \
+      WEBSITE_TAR=/io/www.tar.gz \
+      sh scripts/release.sh
```
[the remark this answers is not in the document]
Comment (ehigham) [reply build.yaml:3386]: where's the comment from dan?

[the remark this answers is not in the document]
Comment (daniel-goldstein) [reply build.yaml:3386]: https://github.com/hail-is/hail/blob/3dfb03cd6a01ccaa85a7ae1bf2f4efa2029e9c05/hail/scripts/release.sh#L45

[the remark this answers is not in the document]
Comment (ehigham) [reply build.yaml:3386]: That's just tesing if the env var is empty?

[the remark this answers is not in the document]
Comment (ehigham) [reply build.yaml:3386]: oh I see, you're saying this is not POSIX compatible so should invoke with bash

[the remark this answers is not in the document]
Comment (daniel-goldstein) [reply build.yaml:3386]: Ya, sorry I wasn't super direct. fwiw I'm sure `sh` is bash in this image, but would rather avoid the footgun and utilize the shebang.

[the remark this answers is not in the document]
Comment (ehigham) [reply build.yaml:3386]: Yes it is but that's a fair point

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**